SearchClient/BingWebSearchAdapter.php

```diff
@@ -0,0 +1,285 @@
+<?php
+
+/*
+ * This file is part of the LiipSearchBundle
+ *
+ * (c) Liip AG
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Liip\SearchBundle\SearchClient;
+
+use Liip\SearchBundle\Exception\SearchException;
+use Pagerfanta\Adapter\AdapterInterface;
+
+/**
+ * Adapter for Bing web search API.
+ */
+class BingWebSearchAdapter implements AdapterInterface
+{
+    /**
+     * @var string
+     */
+    private $apiKey;
+
+    /**
+     * @var
+     */
+    private $apiUrl;
+
+    /**
+     * @var string
+     */
+    private $query;
+
+    /**
+     * @var string[]|array
+     */
+    private $restrictToSites = [];
+
+    /**
+     * @var int Number of seconds to wait for api timeout
+     */
+    private $maxTimeoutMs;
+
+    /**
+     * @var bool Number of results found.
+     */
+    private $totalResults = false;
+
+    /**
+     * @var int Current number of retries
+     */
+    private $retryCount;
+
+    /**
+     * @var int Number of seconds to delay before retry
+     */
+    private $retryDelayMs;
+
+    /**
+     * @var int Maximum number of retries
+     */
+    private $maxRetries;
+
+    /**
+     * @param string      $apiKey
+     * @param string      $apiUrl
+     * @param string      $query
+     * @param array       $restrictToSites
+     * @param int         $maxTimeoutMs
+     */
+    public function __construct(
+        $apiKey,
+        $apiUrl,
+        $query,
+        array $restrictToSites = [],
+        $maxTimeoutMs = 1000,
+        $maxRetries = 3,
+        $retryDelayMs = 1000
+    )
+    {
+        $this->apiKey = $apiKey;
+        $this->apiUrl = $apiUrl;
+        $this->query = $query;
+        $this->restrictToSites = $restrictToSites;
+        $this->maxTimeoutMs = $maxTimeoutMs;
+        $this->maxRetries = $maxRetries;
+        $this->retryDelayMs = $retryDelayMs;
+    }
+
+    /**
+     * Get search results
+     *
+     * {@inheritdoc}
+     */
+    public function getSlice($offset, $length)
+    {
+        $results = $this->executeSearch($offset, $length);
+
+        // FIXME: Hopefully this doesnt hide errors
```

Comment:
@dbu It seems this is just indicative that there were no results returned and not that there is an error. The code already handles HTTP non-200 status codes which appears to be how bing's api handles errors and not with the content of the response so I believe this is actually the correct behavior. This is less of a fixme and more of a note that I am uncertain if there is a scenario where this might indicate and error instead of 'no results'.